EhsanAsh/employee-tracker
# # Initialization of Management DB: Creation and Seeding Scripts

## Introduction
>This pull request introduces the initial set of SQL scripts for creating and seeding the `management_db` database. The scripts are divided into two distinct files: `schema.sql` for database and table creation, and `seeds.sql` for populating the tables with initial data.

## File Breakdown

#### `schema.sql` Overview
- **Database Creation**: Initiates by checking and dropping the existing `management_db` if present, followed by creating a new one.
- **Table Structures**: 
  - `department` Table: Defined to store department names with unique IDs and timestamps.
  - `role` Table: Contains role titles, associated salaries, and a foreign key reference to the department each role belongs to, maintaining referential integrity.
  - `employee` Table: Designed to store employee data, including names, role IDs, and manager IDs with appropriate foreign key constraints for data consistency and integrity.

#### `seeds.sql` Overview
- **Data Seeding**:
  - `department` Table: Populated with a variety of departments, each identified by a unique name.
  - `role` Table: Seeded with a list of roles, each associated with a salary and a department ID.
  - `employee` Table: Inserted with employee data, ensuring each employee is associated with a valid role and manager (if applicable).

## Testing Instructions
- Execute the `schema.sql` script to establish the database and its tables.
- Run the `seeds.sql` script to populate the tables with the initial set of data.
- Perform SELECT queries on each table to ensure data is inserted appropriately and relationships among the tables are maintained accurately.

## Points to Consider
- When running `seeds.sql`, ensure that each `role_id` in employee insertion statements corresponds to a valid `id` in the `role` table.
- Insert manager records first, followed by their respective subordinates to maintain accurate `manager_id` references in the `employee` table.

## Conclusion
This initial commit lays the foundation of the `management_db` database by setting up the necessary tables and seeding them with a preliminary set of data, providing a solid base for future development and data manipulation tasks within the database.
